presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetPageReader.java

```diff
@@ -111,4 +111,9 @@ public ParquetDictionaryPage readDictionaryPage()
             throw new RuntimeException("Error reading dictionary page", e);
         }
     }
+
+    public boolean available()
```

Comment:
I feel like we don't need this `available()` method. The `readPage()` method returns `null` when there are no pages to read. So, we can detect this condition by looking at the returned value from `readPage()`.

I made some changes to the read loop to get rid of `available()` and added some assertions, please take a look: https://gist.github.com/nezihyigitbasi/efc948726137f917af671ed7b9923ce9